flaml/model.py

```diff
@@ -856,20 +850,11 @@ def predict(self, X, **pred_kwargs):
                 test_dataset,
                 metric_key_prefix="predict",
             )
-
-        if self._task == SEQCLASSIFICATION:
-            return np.argmax(predictions.predictions, axis=1)
-        elif self._task == SEQREGRESSION:
-            return predictions.predictions.reshape((len(predictions.predictions),))
-        elif self._task == TOKENCLASSIFICATION:
-            return np.argmax(predictions.predictions, axis=2)
-        elif self._task == SUMMARIZATION:
-            decoded_preds = self.tokenizer.batch_decode(
-                predictions.predictions, skip_special_tokens=True
-            )
-            return decoded_preds
-        elif self._task == MULTICHOICECLASSIFICATION:
-            return np.argmax(predictions.predictions, axis=1)
+        test = postprocess_prediction(
+            self._task, predictions.predictions, self.tokenizer
+        )

```

Comment:
why don't we need to postprocess seq2seq predictions here?

Reply:
because the function doesn't have access to the label here